### PR TITLE
Add markdown editor on conversations and comments

### PR DESCRIPTION
**Pull request by cmrd-senya**
This is a follow up for #7214.

I fixed all the notes by @svbergerem and rebased the code.

I didn't do thorough review or test myself, so this needs review. Maybe I'll do one myself if nobody else does it earlier.

### CODE REVIEW

**Comment by goobertron** [comment]
Great, thanks!

**Comment by cmrd-senya** [comment]
Okay, I reviewed the code and made some changes I found desirable. For me it looks good now. I think we need a one more review now, possibly from @svbergerem to merge it.

File: spec/javascripts/helpers/markdown_editor_spec.js [hunk 1]
```diff
@@ -220,6 +220,38 @@ describe("Diaspora.MarkdownEditor", function() {
...
+  describe("isPreviewMode", function() {
+    beforeEach(function() {
+      this.target = new Diaspora.MarkdownEditor(this.$el, {onPreview: $.noop, onPostPreview: $.noop()});
+    });
+
+    it("return false if editor is not visible yet", function() {
+      this.target.instance = undefined;
+      expect(this.target.isPreviewMode()).toBe(false);
+    });
+
```
Comment (svbergerem) [inline spec/javascripts/helpers/markdown_editor_spec.js]: Please add `returns false if the editor is in write mode`.

File: spec/javascripts/helpers/markdown_editor_spec.js [hunk 2]
```diff
@@ -220,6 +220,38 @@ describe("Diaspora.MarkdownEditor", function() {
...
+  describe("userInputNotEmpty", function() {
+    beforeEach(function() {
+      this.target = new Diaspora.MarkdownEditor(this.$el, {onPreview: $.noop, onPostPreview: $.noop()});
+    });
+
+    it("return false if editor is not visible yet", function() {
+      this.target.instance = undefined;
+      expect(this.target.userInputNotEmpty()).toBe(false);
+    });
+
```
Comment (svbergerem) [inline spec/javascripts/helpers/markdown_editor_spec.js]: As written above, I think this method should be called `userInputEmpty` and behave accordingly. Afterwards please add a test `returns true if the editor is empty`.

**Review by svbergerem — CHANGES_REQUESTED** [review]
Just some minor remarks. Everything else looks good. Please squash the commits afterwards.

**Comment by cmrd-senya** [comment]
I squashed and rebased everything that was done before your notes. For your notes I added changes as a fixup commit so you could check what I changed exactly.

**Review by svbergerem — APPROVED** [review]

**Comment by svbergerem** [comment]
I squashed the fixup commit while merging. Thank you! :)

**Comment by Flaburgan** [comment]
Just pushed that change to diaspora-fr. Thank you two VERY MUCH for your work!

**Comment by goobertron** [comment]
Yes, a HUGE thank you to @cmrd-senya and @svbergerem!

**Comment by cmrd-senya** [comment]
And to @AugierLe42e who developed the original code!
